src/viking.js

```diff
@@ -1,11 +1,96 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(theDamage) {
+    this.theDamage = theDamage;
+    this.health = this.health - theDamage;
+    if (this.health > 0) {
+      return `${this.name} has received ${this.theDamage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(theDamage) {
+    this.theDamage = theDamage;
+    this.health = this.health - theDamage;
+    if (this.health > 0) {
+      return `A Saxon has received ${this.theDamage} points of damage`;
+    } else {
+      return `A Saxon has died in combat`;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  addViking(viking) {
+    this.vikingArmy.push(viking);
+  }
+
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon);
+  }
+  vikingAttack() {
+    let result = this.saxonArmy[this.saxonArmy.length - 1].receiveDamage(
+      this.vikingArmy[this.vikingArmy.length - 1].strength
+    );
+
+    if (this.saxonArmy[this.saxonArmy.length - 1].health <= 0) {
+      this.saxonArmy.pop();
+    }
+    return result;
+  }
```

Comment:
Works well, but in that case it's always the last viking of the vikingArmy array attacking the last saxon of the saxonArmy array 😉 You could try choosing a saxon and viking at random like specified in the lab instructions (although the jasmine tests were not checking it)